deployment/postgres-init-db/sql/tables/merlin/constraints/constraint_results.sql

```diff
@@ -0,0 +1,45 @@
+create table merlin.constraint_results (
+  id integer not null generated always as identity,
+
+  -- inputs
+  constraint_id integer not null,
+  constraint_revision integer not null,
+  simulation_dataset_id integer not null,
+  arguments jsonb not null,
+
+  -- outputs
+  results jsonb not null default '{}',
+  errors jsonb not null default '{}',
```

Comment:
Just for my understanding - is the expectation that only one of `results` or `errors` will be populated for a given row? Would documenting the answer to that question in a brief comment in this sql file be worthwhile?

Reply:
That's how the table is written to, and I'm trying to think of a case where you'd want to write to both. I hadn't considered it a restriction of the table, though, which is why there's no constraint or comment.